HeterogeneousCore/SonicTriton/src/TritonData.cc

```diff
@@ -116,14 +130,16 @@ void TritonInputData::toServer(std::shared_ptr<TritonInput<DT>> ptr) {
   //shape must be specified for variable dims or if batch size changes
   data_->SetShape(fullShape_);
 
-  if (byteSize_ != sizeof(DT))
-    throw cms::Exception("TritonDataError") << name_ << " input(): inconsistent byte size " << sizeof(DT)
+  std::unique_ptr<TritonConverterBase<DT>> converter = createConverter<DT>();
+
+  if (byteSize_ != converter->getByteSize())
```

Comment:
I'm not sure if this change is correct. The original purpose here was a runtime check that the type in the user's code (`DT`) corresponds to the model specification on the server (`byteSize_`). The converter can have a different byte size in the FPGA case, but is that what the server will return from the model specification?

Reply:
I think this is necessary. I think the model specification on the server should use a type with the appropriate bitwidth (i.e. for the ap_fixed<16,6> FPGA setup the server returns UINT16 for the data type, and for a FP16 GPU model I think something similar would get done). But the user wants to just load the data as float32s in these cases.